src/Fixer/Casing/MagicConstantCasingFixer.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Casing;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author ntzm
+ */
+final class MagicConstantCasingFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $magicConstants = $this->getMagicConstants();
+        $magicConstantTokens = $this->getMagicConstantTokens();
+
+        for ($index = 0, $count = $tokens->count(); $index < $count; ++$index) {
```

Comment:
this can be simplified even further now :)
```php
foreach($tokens as $token) {
    if ($token->isGivenKind($magicConstantTokens)) {
        $token->setContent($magicConstants[$token->getId()]);
    }
}
```